### PR TITLE
feat(whiteboard) add user limit

### PR DESCRIPTION
**Pull request by mihhu**

### CODE REVIEW

File: react/features/whiteboard/middleware.ts [hunk 1]
```diff
@@ -76,6 +81,9 @@ MiddlewareRegistry.register((store: IStore) => (next: Function) => async (action
         }
 
         if (action.isOpen) {
+            if (shouldEnforceUserLimit(state)) {
```
Comment (saghul) [inline react/features/whiteboard/middleware.ts:84]: Shouldn't we do this above too, where we setup the whiteboard?

**Review by saghul — COMMENTED** [review]
If a user clicks on the whiteboard button after the limit has been reached they should see an explanatory dialog.

File: react/features/whiteboard/middleware.ts (hunk 1 above)
Comment (mihhu) [reply react/features/whiteboard/middleware.ts:84]: If you don't have collab details, you're the first one to open it. The rest receive the collab details and go through this if. Or were you thinking about something else?

**Comment by mihhu** [comment]
> If a user clicks on the whiteboard button after the limit has been reached they should see an explanatory dialog.

Yes, I totally agree, I just thought you wanted me to do that afterwards. I'll add it in here.

File: react/features/whiteboard/middleware.ts (hunk 1 above)
Comment (saghul) [reply react/features/whiteboard/middleware.ts:84]: Right, makes sense!

Comment (mihhu) [reply react/features/whiteboard/middleware.ts:84]: Maybe I should take into account participant_left events and bump the other participants to the whiteboard if there's any room left. But this splits the participant into those who have access and those who don't, so would it be ok to keep an array of jids in the whiteboard store?

Comment (mihhu) [reply react/features/whiteboard/middleware.ts:84]: *Only if the userLimit is present, otherwise it wouldn't make sense to keep all the jids like that

Comment (saghul) [reply react/features/whiteboard/middleware.ts:84]: Nah, they can press the button if they want to, IMHO. Let's not overcomplicate it for now.

**Comment by saghul** [comment]
> > If a user clicks on the whiteboard button after the limit has been reached they should see an explanatory dialog.
> 
> Yes, I totally agree, I just thought you wanted me to do that afterwards. I'll add it in here.

We can do it in a subsequent PR, no problem.

**Review by saghul — DISMISSED** [review]

File: config.js [hunk 2]
```diff
@@ -1685,6 +1685,11 @@ var config = {
     //     // The server used to support whiteboard collaboration.
     //     // https://github.com/jitsi/excalidraw-backend
     //     collabServerBaseUrl: 'https://excalidraw-backend.example.com',
+    //     // The user access limit to the whiteboard, introduced as a means
+    //     // to control the performance.
+    //     userLimit: 25,
+    //     // The url for more info about the whiteboard and its usage limitations.
+    //     limitUrl: 'https://example.com/blog/whiteboard-limits
```
Comment (saghul) [inline config.js:1692]:
```suggestion
    //     limitUrl: 'https://example.com/blog/whiteboard-limits',
```

File: react/features/whiteboard/components/web/WhiteboardLimitDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @returns {JSX.Element}
+ */
+const WhiteboardLimitDialog = () => {
+    const { t } = useTranslation();
+    const { limitUrl } = useSelector(getWhiteboardConfig);
+
+    return (
+        <Dialog
+            cancel = {{ translationKey: 'dialog.Ok' }}
```
Comment (saghul) [inline react/features/whiteboard/components/web/WhiteboardLimitDialog.tsx:19]: Why not hide cancel instead?

File: react/features/whiteboard/constants.ts [hunk 4]
```diff
@@ -52,3 +52,8 @@ export const WHITEBOARD_UI_OPTIONS = {
         }
     }
 };
+
+/**
+ * Whiteboard soft limit factor.
+ */
+export const USER_LIMIT_THRESHOLD = 0.25;
```
Comment (saghul) [inline react/features/whiteboard/constants.ts:59]: Not sure a percentage is a good idea here. I'd say we set it to 5 and that's that. If you set it to 100 we'd be acommodating too many more for example.

**Review by saghul — COMMENTED** [review]
Great work! A couple of minor comments!

File: react/features/whiteboard/constants.ts (hunk 4 above)
Comment (mihhu) [reply react/features/whiteboard/constants.ts:59]: What if the limit is below 5?

Comment (saghul) [reply react/features/whiteboard/constants.ts:59]: That makes no sense, we should not support anything less than 10.

Comment (saghul) [reply react/features/whiteboard/constants.ts:59]: Yes, let's limit the lower boud to 10.

**Review by saghul — APPROVED** [review]
🚀  Excellent work!

**Pull request merged**